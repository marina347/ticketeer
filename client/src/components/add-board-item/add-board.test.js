import React from "react";
import { shallow } from "enzyme";

import { AddBoardItem } from "./add-board-item-form.component";

describe("AddBoard component", () => {
  let wrapper;
  let mockProps;
  let addBoard;
  let event;
  let closePopup;
  let handleSubmitSpy;
  let handleChangeSpy;

  beforeEach(() => {
    addBoard = jest.fn();
    closePopup = jest.fn();
    event = {
      preventDefault: jest.fn(),
      target: {
        name: "boardName",
        value: "testname",
      },
    };
    mockProps = {
      addBoard,
      closePopup,
    };

    //spyes will work if in class declaration you use bind in constructor. on anonymous method spyes cant work
    handleSubmitSpy = jest.spyOn(AddBoardItem.prototype, "handleSubmit");
    handleChangeSpy = jest.spyOn(AddBoardItem.prototype, "handleChange");

    wrapper = shallow(<AddBoardItem {...mockProps} />);
  });
  it("Should render AddBoard component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should start onSubmit method when form inside is submitted", () => {
    wrapper.find('[id="add_board_form"]').simulate("submit", event);
    expect(handleSubmitSpy).toHaveBeenCalled();
    expect(addBoard).toHaveBeenCalled();
    expect(wrapper.state()).toEqual({ boardName: "", boardDescription: "" });
  });

  it("Should set boardName to testname if handleChange is fired", () => {
    wrapper.instance().handleChange(event);
    expect(wrapper.state()).toEqual({
      boardName: "testname",
      boardDescription: "",
    });
  });

  it("Form input has a prop handleChange", () => {
    expect(
      wrapper
        .find('[id="add_board_form"]')
        .find('[id="add_board_input"]')
        .prop("handleChange")
    ).not.toEqual(undefined);
  });

  it("Form area has a prop handleChange", () => {
    expect(
      wrapper
        .find('[id="add_board_form"]')
        .find('[id="add_board_description"]')
        .prop("onChange")
    ).not.toEqual(undefined);
  });
});
