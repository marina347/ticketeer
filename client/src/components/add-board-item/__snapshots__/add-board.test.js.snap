// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBoard component Should render AddBoard component 1`] = `
<form
  class="add-form"
  id="add_board_form"
  onSubmit={[Function]}
>
  <h3
    className="add-form__heading"
  >
    ADD NEW BOARD
  </h3>
  <FormInput
    handleChange={[Function]}
    id="add_board_input"
    label="Board name"
    maxlength="50"
    name="boardName"
    placeholder="Board name"
    required={true}
    type="text"
    value=""
  />
  <FormTextArea
    additionalStylesApplied={true}
    cols={20}
    id="add_board_description"
    label="Board description"
    maxlength="300"
    name="boardDescription"
    onChange={[Function]}
    placeholder="Board description"
    rows={3}
    value=""
  />
  <FormButton
    className="btn btn-main btn-main--gradient u-animation-none"
    type="submit"
  >
    ADD
  </FormButton>
</form>
`;
